### PR TITLE
Enable parquet suites from Spark UT

### PR DESCRIPTION
**Pull request by Feng-Jiang28**
This pr is for enabling parquet suites for UT framework. I used some Gluten test cases in the suite as the UT frame used to.
Besides that, there are more suites and Gluten doesn't cover and Spark 3.3.0 has.
Enabled suites include:
1. RapidsParquetAvroCompatibilitySuite
2. RapidsParquetColumnIndexSuite
3.  RapidsParquetCompressionCodecPrecedenceSuite
4. RapidsParquetDeltaByteArrayEncodingSuite
5. RapidsParquetDeltaEncodingInteger
6. RapidsParquetDeltaEncodingLong
7. RapidsParquetDeltaLengthByteArrayEncodingSuite
8. RapidsParquetFieldIdIOSuite
9. RapidsParquetFieldIdSchemaSuite
10. RapidsParquetFileFormatSuite
11. RapidsParquetInteroperabilitySuite
12. RapidsParquetProtobufCompatibilitySuite
13. RapidsParquetPartitionDiscoverySuite
14. RapidsParquetQuerySuite
15. RapidsParquetRebaseDatetimeSuite
16. RapidsParquetSchemaPruningSuite
17. RapidsParquetSchemaSuite
18. RapidsParquetThriftCompatibilitySuite
19. RapidsParquetVectorizedSuite

V1 and V2 versions of the data source API related suites were not enabled or put in the tests.

### CODE REVIEW

File: tests/pom.xml [hunk 1]
```diff
@@ -156,6 +157,43 @@
                     <version>3.1.0.0-RC2</version>
                     <scope>test</scope>
                 </dependency>
+                <dependency>
+                    <groupId>org.apache.parquet</groupId>
```
[the remark this answers is not in the document]
Comment (Feng-Jiang28) [reply tests/pom.xml:160]: I made changes of the code by removing the parquet related dependencies while keeping parquet-column. I am not sure if I chose the rigtht dependency. Can you please have a look?

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsTestSettings.scala [hunk 2]
```diff
@@ -77,5 +77,18 @@ class RapidsTestSettings extends BackendTestSettings {
     .exclude("SPARK-22550: Elt should not generate codes beyond 64KB", WONT_FIX_ISSUE("https://github.com/NVIDIA/spark-rapids/issues/10775"))
     .exclude("SPARK-22603: FormatString should not generate codes beyond 64KB", WONT_FIX_ISSUE("https://github.com/NVIDIA/spark-rapids/issues/10775"))
   enableSuite[RapidsStringFunctionsSuite]
+  enableSuite[RapidsParquetColumnIndexSuite]
+  enableSuite[RapidsParquetCompressionCodecPrecedenceSuite]
+    .exclude("Create parquet table with compression", KNOWN_ISSUE("To be added"))
```
Comment (jlowe) [inline tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsTestSettings.scala:82]: Tracking issue needs to be filed and noted here.

**Review by jlowe — COMMENTED** [review]
Dependencies looks a lot better.  Need to file the tracking issue but otherwise lgtm.

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Review by thirtiseven — COMMENTED** [review]
Are `output4.parquet` and `.output4.parquet.crc` added on purpose?

**Comment by Feng-Jiang28** [comment]
build

File: scala2.13/tests/pom.xml [hunk 3]
```diff
@@ -156,6 +156,19 @@
...
+                    <groupId>org.apache.parquet</groupId>
+                    <artifactId>parquet-column</artifactId>
+                    <version>${parquet.hadoop.version}</version>
+                    <scope>test</scope>
+                    <classifier>tests</classifier>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.parquet</groupId>
+                    <artifactId>parquet-avro</artifactId>
+                    <version>1.12.2</version>
```
Comment (binmahone) [inline scala2.13/tests/pom.xml:169]: use variable substitution?

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsParquetColumnIndexSuite.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  private val actions: Seq[DataFrame => DataFrame] = Seq(
+    "_1 = 500",
+    "_1 = 500 or _1 = 1500",
+    "_1 = 500 or _1 = 501 or _1 = 1500",
+    "_1 = 500 or _1 = 501 or _1 = 1000 or _1 = 1500",
+    "_1 >= 500 and _1 < 1000",
+    "(_1 >= 500 and _1 < 1000) or (_1 >= 1500 and _1 < 1600)"
+  ).map(f => (df: DataFrame) => df.filter(f))
+
+  testRapids("test reading unaligned pages - test all types") {
```
Comment (binmahone) [inline tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsParquetColumnIndexSuite.scala:36]: nothing is exluded from RapidsParquetColumnIndexSuite?

**Review by binmahone — CHANGES_REQUESTED** [review]

**Comment by Feng-Jiang28** [comment]
build

**Comment by Feng-Jiang28** [comment]
build

**Review by thirtiseven — COMMENTED** [review]
LGTM but not a pom change expert

**Review by binmahone — APPROVED** [review]

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**